src/commonMain/kotlin/exchange.dydx.abacus/validator/TriggerOrdersInputValidator.kt

```diff
@@ -543,6 +621,19 @@ internal class TriggerOrdersInputValidator(
         return null
     }
 
+    private fun requiredTriggerToLiquidationPrice(type: String?, side: String?): RelativeToPrice? {
+        return when (type) {
+            "STOP_MARKET" ->
```

Comment:
If we make these case enums instead of strings, we can avoid having an `else -> null` case and remove a source of nullability.

I would also consider whether `else` is even a valid state to be in. If not, we should just `error()` out instead of returning null. In general, we should be looking to avoid nullable types as much as possible.

Reply:
ah interesting; yeah the current pattern I've seen in the repo is to return early (return null) if we get an unexpected null type (but then detekt yelled at me because we had too many return statements in a function oops). Happy to pivot to using `error()` more liberally - I think this whole file could probably be revamped a bit more (i basically did the min to get it to pass detekt) - filing a tix [here](https://linear.app/dydx/issue/CT-786/clean-up-triggerordersinputvalidatorkt-file).